### PR TITLE
CircleCI Configuration

### PR DESCRIPTION
**Pull request by nhakmiller**
### Background

Until now, CircleCI has not been running. This PR updates the CircleCI configuration to run the tests as appropriate. Additionally this update addresses all the issues reported by the tests that we were previously ignoring because there was non CircleCI in place.

### Changes

* Updated `.circleci/cocnfig` to actually run tests
* Address issues reported by bandit and mypy

### Testing

* Locally

### CODE REVIEW

File: panther_cli/main.py [hunk 1]
```diff
@@ -15,15 +15,16 @@
 '''
 
 import argparse
-from collections import defaultdict
+# from collections import defaultdict
```
Comment (rleighton) [inline panther_cli/main.py]: Are you leaving this commented for a reason?

Comment (nhakmiller) [reply panther_cli/main.py]: Nope, good catch. Removed.

File: .circleci/config.yml [hunk 2]
```diff
@@ -7,5 +7,14 @@ jobs:
     steps:
         - checkout
         - run:
-            name: Pass
-            command: echo 'true'
+            name: Install core utilities
+            command: make install
+        - run:
+            name: Setup the Virtual Environment
+            command: make venv && source venv/bin/activate
```
Comment (austinbyers) [inline .circleci/config.yml]: Not sure if sourcing/pipenv will work the way you expect in CodeBuild. If so, great! If not, you can always run the binaries directly, for example `venv/bin/pylint`

Comment (nhakmiller) [reply .circleci/config.yml]: Well I'm not sure if it works as I expects, but its passing! I"ll leave it as is for now until we have a reason to change.

**Pull request merged**